### PR TITLE
usb-gadget-service: Add USB gadget systemd service using libusbgx and gt

### PR DESCRIPTION
**Pull request by mhennerich**
To install this on your Kuiper Linux image execute install_gt.sh
script in the local folder. These tools and services will be integrated
in the next Kuiper Linux release.

Signed-off-by: Michael Hennerich <michael.hennerich@analog.com>

### CODE REVIEW

File: usb-gadget-service/udev/99-udc.rules [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#  SPDX-License-Identifier: MIT or LGPL-2.1+
+#
+# Copyright 2019 Collabora Ltd
```
Comment (nunojsa) [inline usb-gadget-service/udev/99-udc.rules:3]: Is this on purpose (just double checking?

**Review by nunojsa — APPROVED** [review]

File: usb-gadget-service/udev/99-udc.rules (hunk 1 above)
Comment (mhennerich) [reply usb-gadget-service/udev/99-udc.rules:3]: I copied the file from here:
https://github.com/linux-usb-gadgets/gt

Comment (nunojsa) [reply usb-gadget-service/udev/99-udc.rules:3]: ack

**Comment by mhennerich** [comment]
Tested on RPI4
Added: dtoverlay=dwc2 to config.txt and modules-load=dwc to cmdline.txt

```

 michael@mhenneri-D06:~$ iio_info -s
 Library version: 1.0 (git tag: dc10ecab)
 Compiled with backends: local xml ip usb
 Unable to create Local IIO context : No such file or directory (2)
 Available contexts:
 	0: 10.44.3.83 on Raspberry Pi 4 Model B Rev 1.4 [ip:analog.local]
 	1: 0456:b671 (Analog Devices Inc. IIO-USB-CDC-ACM), serial= [usb:1.42.0]
```

ttyACM0 and IIOD via USB works correctly...

**Review by amiclaus — COMMENTED** [review]
Tested on Rpi Zero 2:

image
image

**Comment by mhennerich** [comment]
Also tested on Zynq-7000

```
michael@mhenneri-D06:~$ iio_info -s
Library version: 1.0 (git tag: dc10ecab)
Compiled with backends: local xml ip usb
Unable to create Local IIO context : No such file or directory (2)
Available contexts:
	0: 10.44.3.71 (AD-FMCOMMS4-EBZ on Xilinx Zynq ZED), serial=00230 [ip:analog.local]
	1: 0456:b671 (Analog Devices Inc. IIO-USB-CDC-ACM), serial= [usb:1.58.0]

```

**Review by dbogdan — APPROVED** [review]

**Pull request merged**